libraries/chain/steem_evaluator.cpp

```diff
@@ -2506,14 +2505,18 @@ void delegate_vesting_shares(database& _db, const chain_properties& median_props
                 GOLOS_THROW_MISSING_OBJECT("vesting_delegation_object", op.delegator);
             }
 
-            _db.modify(delegator, [&](account_object& a) {
-                a.delegated_vesting_shares -= delegation->vesting_shares;
-            });
+            auto now = _db.head_block_time();
 
             _db.modify(delegatee, [&](account_object& a) {
                 a.received_vesting_shares -= delegation->vesting_shares;
             });
 
+            _db.create<vesting_delegation_expiration_object>([&](vesting_delegation_expiration_object& o) {
+                o.delegator = op.delegator;
+                o.vesting_shares = delegation->vesting_shares;
+                o.expiration = now + STEEMIT_CASHOUT_WINDOW_SECONDS;
```

Comment:
Change to full code:
```
o.expiration = std::max(now + STEEMIT_CASHOUT_WINDOW_SECONDS, delegation->min_delegation_time);
```
It prevents abusing `create_account_with_delegation` to create own accounts and return delegation faster then allowed.

Maybe move whole _db.create to separate helper function, used by reject and `delegate_vesting_shares_evaluator`